contribs/gmf/src/directives/authenticationdirective.js

```diff
@@ -0,0 +1,281 @@
+goog.provide('gmf.AuthenticationController');
+goog.provide('gmf.authenticationDirective');
+
+goog.require('gmf');
+goog.require('gmf.Authentication');
+goog.require('ngeo.modalDirective');
+
+
+gmfModule.value('gmfAuthenticationTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfAuthenticationTemplateurl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/authentication.html';
+    });
+
+
+/**
+ * An "authentication" directive for a GeoMapFish application. With the
+ * use of the "authentication" service, it features a complete interface
+ * for the user to be able to login, logout, change or reset his or her
+ * password.  The `gmfUser` angular value is also used to keep track of
+ * the user information. When empty, that means that the user isn't connected
+ * yet.
+ *
+ * While not logged in, the "login" form is shown, which allows the user to
+ * either log in or ask for a password reset.
+ *
+ * Once logged in, the "logout" form is shown, which allows the user to either
+ * log out or change his or her password.
+ *
+ * @example
+ * <gmf-authentication></gmf-authentication>
+ *
+ * @param {string} gmfAuthenticationTemplateUrl Url to template.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfAuthentication
+ */
+gmf.authenticationDirective = function(gmfAuthenticationTemplateUrl) {
+  return {
+    scope: {},
+    controller: 'GmfAuthenticationController',
+    controllerAs: 'authCtrl',
+    templateUrl: gmfAuthenticationTemplateUrl
+  };
+};
+
+gmfModule.directive('gmfAuthentication', gmf.authenticationDirective);
+
+
+
+/**
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {gmf.Authentication} gmfAuthentication GMF Authentication service
+ * @param {gmf.User} gmfUser
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfAuthenticationController
+ */
+gmf.AuthenticationController = function(gettextCatalog, $scope,
+    gmfAuthentication, gmfUser) {
+
+  $scope.gmfUser = gmfUser;
```

Comment:
I remember now.  I set it in the `$scope` to be able to do the following in the template:

    <div ng-if="gmfUser.username">

I haven't found a way to use an angular value inside a template otherwise, so far.  Is there a better way to accomplish this without having to set it in the scope ?